packages/beacon-node/src/network/reqresp/handlers/dataColumnSidecarsByRange.ts

```diff
@@ -57,17 +66,21 @@ export async function* onDataColumnSidecarsByRange(
         // Spec: https://github.com/ethereum/consensus-specs/blob/ad36024441cf910d428d03f87f331fbbd2b3e5f1/specs/fulu/p2p-interface.md#L425-L429
         const dataColumnSidecars = await unfinalized.getManyBinary(fromHex(block.blockRoot), columns);
         for (const [index, dataColumnSidecarBytes] of dataColumnSidecars.entries()) {
-          if (!dataColumnSidecarBytes) {
-            throw new ResponseError(
-              RespStatus.SERVER_ERROR,
-              `No unfinalized dataColumnSidecar found for root=${block.blockRoot}, index=${columns[index]}`
-            );
+          if (dataColumnSidecarBytes) {
+            yield {
+              data: dataColumnSidecarBytes,
+              boundary: chain.config.getForkBoundaryAtEpoch(computeEpochAtSlot(block.slot)),
+            };
+          } else {
+            await logDataColumnSidecarUnavailability({
```

Comment:
all the logs added in this PR will mostly be useful to help diagnose bugs in other clients, would be good to have the peer id as part of the log (and maybe even client name)